src/rgw/rgw_asio_frontend.cc

```diff
@@ -138,6 +138,7 @@ void handle_connection(boost::asio::io_context& context,
                        boost::system::error_code& ec,
                        spawn::yield_context yield)
 {
+  init_jaeger();
```

Comment:
why do you init here?
is it done for reconnections?

Reply:
init means connecting jaeger backend cant do it in common_init as it runs just one time when server starts, internally `init` checks if the server is already been initialized if yes it wont initialize again

Reply:
so this is for retry in case the initial connection failed?
what about the case that the initial connection was ok, but disconnected after that?
is there an internal reconnect mechanism for opentracer?

Reply:
Nope this mechanism is not there, jaeger uses UDP connection to pass span
data to server so reliability is a issue here

Reply:
if we do not have a reconnect mechanism, why do we need to call init on every request?

Reply:
actually its purpose is to be called once only, but I cant put this inside common_init.cc because this will run only during start and if by any chance it misses to connect at that time then the whole server is needed to be restarted again for reconnection,
puttin it inside asio_frontend because in this way there will be a way to check for connection if not connected on everry request, and you know a atomic bool flag is already there to check if it is connected inside init_tracer() method

Reply:
> actually its purpose is to be called once only, but I cant put this inside common_init.cc because this will run only during start and if by any chance it misses to connect at that time then the whole server is needed to be restarted again for reconnection,

if the transport is over UDP, what would happen during the connection process?

> puttin it inside asio_frontend because in this way there will be a way to check for connection if not connected on everry request, and you know a atomic bool flag is already there to check if it is connected inside init_tracer() method

note that the atomic bool does not protect you from race condition when multiple call to asio_frontend are done at the same time in different threads.